### PR TITLE
Nrh 12 organization revenue

### PR DESCRIPTION
**Pull request by JeremyGibson**
Adds the models for Organization, RevenueProgram

Plan and Feature are modeled but may be substantially changed by a forthcoming ticket, but were needed for the relationship to organization.

Branched off of NRH-13

The significant bit of business logic that Michael and I settled on here is the immutability of the Organization name and slug. This also makes a revenue program immutable once it is created and linked to an organization as the the revenue program's slug is based on the organization slug.

- [x] Tests

### CODE REVIEW

File: apps/organizations/admin.py [hunk 1]
```diff
@@ -1,15 +1,62 @@
...
+        ("Plan", {"fields": ("non_profit", "plan", "stripe_account", "salesforce_id")}),
+    )
+
+    readonly_fields = ["slug"]
+
+    list_display = ["name", "slug", "plan", "org_state"]
+
+    list_filter = ["name", "plan", "org_state"]
+
+    fieldsets = organization_fieldset
```
Comment (michael-caktus) [inline apps/organizations/admin.py:29]: Why this pattern?
`fieldsets = [model]_fieldset`?

**Comment by JeremyGibson** [comment]
Its really a just in case thing. Just in case a different set of fields is configured and appended. So you could have fieldsets = organization_fieldsets + runtime_fieldsets

File: apps/organizations/tests/factories.py [hunk 2]
```diff
@@ -1,8 +1,74 @@
...
+        model = models.RevenueProgram
+
+    name = " ".join(fake.words(nb=4))
+    organization = factory.SubFactory(OrganizationFactory)
+
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
+        obj = model_class(*args, **kwargs)
+        name = kwargs.get("name", None)
+        organization = kwargs.get("organization", None)
```
Comment (copelco) [inline apps/organizations/tests/factories.py:68]: Isn't passing related objects supported by default with FactoryBoy? For example: https://github.com/deardurham/dear-petition/blob/master/dear_petition/petition/conftest.py#L44

Comment (JeremyGibson) [reply apps/organizations/tests/factories.py:68]: Cool. I ran into this problem awhile ago and have just been following the pattern. I'll refactor

File: apps/organizations/models.py [hunk 3]
```diff
@@ -1,18 +1,72 @@
...
+    org_city = models.CharField(max_length=64, blank=True, verbose_name="City")
+    org_state = models.CharField(max_length=2, blank=True, choices=STATE_CHOICES, verbose_name="State")
+    org_zip = models.CharField(max_length=9, blank=True, verbose_name="Zip")
+    salesforce_id = models.CharField(max_length=255, blank=True, verbose_name="Salesforce ID")
+
+    def __str__(self):
+        return self.name
+
+    def current_states(self):
+        return
```
Comment (michael-caktus) [inline apps/organizations/models.py:52]: What's this?

Comment (JeremyGibson) [reply apps/organizations/models.py]: removed.

**Review by michael-caktus — CHANGES_REQUESTED** [review]

File: apps/organizations/views.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+from apps.organizations.models import Feature, Organization, Plan, RevenueProgram
+
+
+class OrganizationViewSet(viewsets.ModelViewSet):
+    queryset = Organization.objects.all()
+
+    def get_serializer_class(self):
+        if self.action == "retrieve":
+            return serializers.OrganizationDetailSerializer
+        return serializers.OrganizationListSerializer
```
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply apps/organizations/views.py:13]: Hmm. Any need to leave this open for access through the DRF api browser?

**Review by michael-caktus — APPROVED** [review]

**Pull request merged**